src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11302,15 +11320,55 @@ private bool ScanCast(bool forPattern = false)
 
         private bool IsPossibleLambdaExpression(Precedence precedence)
         {
-            // Only call into this if after `static` or after a legal identifier.
+            // Only call into this if after `static`, '[',  or after a legal identifier.
             Debug.Assert(
-                this.CurrentToken.Kind == SyntaxKind.StaticKeyword ||
+                this.CurrentToken.Kind is SyntaxKind.StaticKeyword or SyntaxKind.OpenBracketToken ||
                 this.IsTrueIdentifier(this.CurrentToken));
             if (precedence > Precedence.Lambda)
             {
                 return false;
             }
 
+            if (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken)
+            {
+                var resetPoint = this.GetResetPoint();
+
+                _ = ParseAttributeDeclarations();
+                bool result;
+
+                switch (CurrentToken.Kind)
+                {
+                    case SyntaxKind.StaticKeyword:
+                    case SyntaxKind.IdentifierToken:
+                        result = IsPossibleLambdaExpressionCore(precedence);
+                        break;
+                    case SyntaxKind.OpenParenToken:
+                        result = ScanParenthesizedImplicitlyTypedLambda(precedence) || ScanExplicitlyTypedLambda(precedence);
+                        break;
+                    default:
+                        result = false;
+                        break;
+                }
+
+                this.Reset(ref resetPoint);
+                this.Release(ref resetPoint);
+
+                return result;
+            }
+            else
+            {
+                return IsPossibleLambdaExpressionCore(precedence);
+            }
+        }
+
+        private bool IsPossibleLambdaExpressionCore(Precedence precedence)
```

Comment:
Consider making this a local function.

Reply:
I considered that but I was reluctant to change the identing for this large method with the merge conflicts that might occur down the road, etc.

---
In reply to: [600895804](https://github.com/dotnet/roslyn/pull/52038#discussion_r600895804) [](ancestors = 600895804)